catalog-rest-service/src/main/resources/ui/src/pages/DashboardDetailsPage/DashboardDetailsPage.component.tsx

```diff
@@ -55,9 +90,24 @@ const DashboardDetailsPage = () => {
   >([]);
 
   const activeTabHandler = (tabValue: number) => {
-    setActiveTab(tabValue);
+    const currentTabIndex = tabValue - 1;
+    if (dashboardDetailsTabs[currentTabIndex].path !== tab) {
+      setActiveTab(getCurrentTab(dashboardDetailsTabs[currentTabIndex].path));
+      history.push({
+        pathname: getDashboardDetailsWithTabPath(
+          dashboardFQN,
+          dashboardDetailsTabs[currentTabIndex].path
+        ),
+      });
+    }
   };
 
+  useEffect(() => {
+    if (dashboardDetailsTabs[activeTab - 1].path !== tab) {
```

Comment:
@ShaileshParmar11 for what purpose we have added this condition here?
same question for other entities as well.

Reply:
if the active tab is the same as `tab` we don't want to update the `setActiveTab`